Automattic/sensei
# Add PHP syntax check CI workflow

Related: https://github.com/Automattic/sensei/pull/7021, https://github.com/Automattic/sensei/pull/6723
Context: p1689270960473579-slack-C02NWDZBL0H

## Proposed Changes

This PR adds a PHP syntax check CI workflow.

It turned out that wordpress.org SVN repo is doing a [pre-commit syntax check on PHP 8.1](https://meta.trac.wordpress.org/ticket/3791#comment:44) and we fail to pass it at times (see related PRs). This change allows us to know if our syntax is good on PHP version 8.1.

The reason for the syntax check failure is related to `php-scoper`. If we compile the dependencies using `php-scoper` on PHP 7.3 (as we do when making the final build), in some cases, the syntax is not valid on PHP 8.1.

## Testing Instructions


1. Check the `PHP Syntax Check` CI workflow and make sure they are passing.
2. Update the `symfony/polyfill-php80` composer package: `composer require --dev symfony/polyfill-php80:1.27.0`
3. Commit the change to this PR.
4. The `PHP Syntax Check` CI workflow should be failing.
5. Revert the change.

## Pre-Merge Checklist
